src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,43 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.*;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type.equals(Bulldozer.class)) {
+            BulldozerProducer bulldozer = new BulldozerProducer();
+            return new ArrayList<>(bulldozer.get());
+        }
+        if (type.equals(Truck.class)) {
+            TruckProducer truck =  new TruckProducer();
+            return new ArrayList<>(truck.get());
+        }
+        if (type.equals(Excavator.class)) {
+            ExcavatorProducer excavator = new ExcavatorProducer();
+            return new ArrayList<>(excavator.get());
+        }
```

Comment:
Consider creating a local variable in your implementation of `getAll(Class type)` method. This will make your code cleaner and more efficient. You can do something like this: `List<? extends Machine> machines;` and then use it in each condition.